### PR TITLE
Separate images for the web and the back-end

### PR DESCRIPTION
**Pull request by tvuillemin**
Instead of a global AppEngine image, there are now two separate docker images for the web front-end and the back-end. 

This PR depends on https://github.com/lumapps/lumsites/pull/6749

### CODE REVIEW

**Comment by volnt** [comment]
This PR confirms that this repository must be private. It's not generic code, it's very lumapps-oriented. @tvuillemin see https://docs.google.com/document/d/1PmHbkOujWYm4g_AmS9mej5j8oo-Mc22x-h_NYxelmmI/edit#heading=h.s19w6nkhcg05 on how to delete github repository

File: back-devserver/run.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+set -e
+
+# Create a dedicated user
+useradd --shell /bin/bash -u ${UID:-1000} -o -c "" -m docker_user
+
+# Install the Python requirements
+gosu docker_user ./utils/install_requirements.sh
```
Comment (volnt) [inline back-devserver/run.sh]: I'm not sure we want to installl the requirements every time we run the container, why aren't we installing them inside the Dockerfile ?

File: web-devserver/run.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+# Copy the SSH credentials
+gosu node mkdir -p /home/node/.ssh/
+gosu node cp /ssh/id_rsa* /home/node/.ssh/
+gosu node cp /ssh/known_hosts /home/node/.ssh/
+chown -R node:node /home/node/.ssh/
+chmod 700 /home/node/.ssh/id_rsa
+
+# Install the web application dependencies
+gosu node yarn install
```
Comment (volnt) [inline web-devserver/run.sh]: Same question here, we aren't we installing the dependencies inside the Dockerfile ?

File: back-devserver/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+
+# Ports declaration
+EXPOSE 8000
+EXPOSE 8888
+
+
+# Runtime
+WORKDIR /app
+COPY run.sh /root/run.sh
```
[the remark this answers is not in the document]
Comment (tgermain) [reply back-devserver/Dockerfile]: Seen together : 
As a first step and while front and back source code live in the same repository we can only work docker images with source code mounted inside as volume.

[the remark this answers is not in the document]
Comment (tgermain) [reply back-devserver/Dockerfile]: Seen together : 
As a first step and while front and back source code live in the same repository we can only work docker images with source code mounted inside as volume.

File: web-devserver/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    rm -rf /var/lib/{apt,dpkg,cache,log}/;
+
+
+# Ports declaration
+EXPOSE 3000
+
+
+# Runtime
+WORKDIR /app
+COPY run.sh /root/run.sh
```
[the remark this answers is not in the document]
Comment (tgermain) [reply web-devserver/Dockerfile]: Seen together : 
As a first step and while front and back source code live in the same repository we can only work docker images with source code mounted inside as volume.

File: web-devserver/run.sh (hunk 2 above)
Comment (tgermain) [reply web-devserver/run.sh]: Seen together : 
As a first step and while front and back source code live in the same repository we can only work docker images with source code mounted inside as volume.

**Pull request merged**